lib/resque/scheduler/delaying_extensions.rb

```diff
@@ -122,15 +119,16 @@ def next_item_for_timestamp(timestamp)
         key = "delayed:#{timestamp.to_i}"
 
         encoded_item = redis.lpop(key)
-        redis.srem("timestamps:#{encoded_item}", key)
-        item = decode(encoded_item)
 
         # If the list is empty, remove it.
         clean_up_timestamp(key, timestamp)
-        item
+
+        decode(encoded_item) if encoded_item
```

Comment:
Is `decode` here `Resque::decode`? Why the new `if` logic here? How did this not blow up before if `encoded_item` was falsy?